core/example/02_tuple/tuple_example.cpp

```diff
@@ -0,0 +1,118 @@
+/****************************************************************************
+ * Copyright (c) 2018 by the Cabana authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cabana_Core.hpp>
+
+#include <iostream>
+
+//---------------------------------------------------------------------------//
+// Tuple example.
+//---------------------------------------------------------------------------//
+void tupleExample()
+{
+    /*
+      Cabana tuples are similar to C++11 tuples in that they are ordered lists
+      of elements of different types specified at compile time. Some aspects
+      of Cabana::Tuple that are different from std::tuple:
+
+      *: Cabana tuples must consist of data of trivial types and classes. A
+         trivial type or class is one who's
+
+         1) storage is contiguous (i.e. trivially copyable)
+
+         2) which only supports static default initialization (trivially
+            default constructible), either cv-qualified or not.
+
+         Trivial types include scalar types, trivial classes and arrays of any
+         such types. A trivial class is a class (defined with class, struct or
+         union) that is both trivially default constructible and trivially
+         copyable, which implies that:
+
+         1) uses the implicitly defined default copy and move constructors,
+            copy and move assignments, and destructor
+
+         2) has no virtual members
+
+         3) has no non-static data members with brace- or equal- initializers
+
+         4) its base class and non-static data members (if any) are themselves
+            also trivial types
+
+      *: The types in a Cabana tuple are defined via a type parameter
+         pack using the Cabana::MemberTypes class
+
+      *: Cabana tuples are designed to store and access multidimensional array
+         data with accessors provided to obtain references to individual array
+         values within the tuple.
+    */
```

Comment:
For the tuple example, It may be worth adding that "Cabana tuple's memory layout is consistent with order of declaration, i.e., what declared first is first in memory. (Note that libstdc++ has reverse order for std::tuple.)